src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -387,4 +405,65 @@ TEST_CASE("request_rate_sequence")
   trrm.TestSequences();
 }
 
+/// Check custom json data to ensure that it is processed correctly
+///
+TEST_CASE("custom_json_data")
+{
+  PerfAnalyzerParameters params;
+
+  SUBCASE("non sequence model")
+  {
+    bool is_seq_model = false;
+    bool using_mock_infer = false;
+    TestRequestRateManager trrm{params, is_seq_model, using_mock_infer};
+    std::unique_ptr<MockDataLoader> mdl = std::make_unique<MockDataLoader>();
+    MockModelParser mmp{false};
+
+    auto thread_status = std::make_shared<TestRequestRateManager::ThreadStat>();
+    thread_status->status_ = cb::Error::Success;
+    thread_status->cb_status_ = cb::Error::Success;
+
+    auto thread_config =
+        std::make_shared<TestRequestRateManager::ThreadConfig>(0, 0);
+    thread_config->non_sequence_data_step_id_ = 0;
+
+
+    std::string ss = R"(
+  {
+      "data" : [
```

Comment:
why list 4 possibilities when they are all the same?

Reply:
Each dictionary (`{...}`) represents one stream, and each sequence being run will attempt to pull from a different stream. So this is kinda testing that multiple streams of data can be provided for multiple sequences, I think.